service/controller/resource/statuscondition/create.go

```diff
@@ -286,22 +286,24 @@ func computeCreatedCondition(status infrastructurev1alpha2.CommonClusterStatus,
 func computeUpdatingCondition(status infrastructurev1alpha2.CommonClusterStatus, desiredVersion string) bool {
 	currentVersion := status.LatestVersion()
 	isCreated := status.HasCreatedCondition()
-	notUpdating := !status.HasUpdatingCondition()
+	notUpdating := status.LatestCondition() != infrastructurev1alpha2.ClusterStatusConditionUpdating
```

Comment:
Here we used to ensure that the cluster was _never_ updating before. Now we ensure that the cluster is not _currently_ updating.